packages/pagination/src/usePagination.ts

```diff
@@ -5,17 +5,45 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { useSelection } from '@zendeskgarden/container-selection';
+import {
+  useSelection,
+  IUseSelectionProps,
+  IUseSelectionState,
+  IGetItemPropsOptions
+} from '@zendeskgarden/container-selection';
 
-export function usePagination(options) {
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface IUsePaginationProps<Item> extends IUseSelectionProps<Item> {}
+
+export interface IGetPageProps<Item> extends IGetItemPropsOptions<Item> {
+  page?: number;
+  current?: any;
+  ariaLabel?: any;
+}
+
+export interface IGetContainerProps extends React.HTMLProps<any> {
+  role?: any;
+  ariaLabel?: string;
+}
+
+export interface IUsePaginationReturnValue<Item> extends IUseSelectionState<Item> {
+  getContainerProps: <T>(options?: T & IGetContainerProps) => any;
+  getPageProps: <T extends IGetPageProps<Item>>(options?: T) => any;
+  getPreviousPageProps: <T extends IGetPageProps<Item>>(options?: T) => any;
+  getNextPageProps: <T extends IGetPageProps<Item>>(options?: T) => any;
+}
+
+export function usePagination<Item>(
```

Comment:
We may want to default this generic to `<Item = any>` to allow usages where people don't want to worry about it.

This would align with `useButtonGroup`

Reply:
Yes, that sounds good. Thanks for the suggestion!